### PR TITLE
fix load weighted prices

### PR DESCRIPTION
**Pull request by lisazeyen**
## Changes proposed in this Pull Request
Fix bug when calculating load weighted prices in the `make_summary` script

## Checklist

- [x] I tested my contribution locally and it works as intended.
- [x] Code and workflow changes are sufficiently documented.
- [x] Changed dependencies are added to `envs/environment.yaml`.
- [x] Changes in configuration options are added in `config/config.default.yaml`.
- [x] Changes in configuration options are documented in `doc/configtables/*.csv`.
- [x] Sources of newly added data are documented in `doc/data_sources.rst`.
- [ ] A release note `doc/release_notes.rst` is added.

### CODE REVIEW

**Comment by lisazeyen** [comment]
@millingermarkus can you check if it is working with this branch?

File: scripts/make_summary.py [hunk 1]
```diff
@@ -540,77 +540,21 @@ def calculate_prices(n, label, prices):
...
-                )
-
-        # Add H2 Store when charging
-        # if carrier == "H2":
-        #    stores = n.stores_t.p[buses+ " Store"].groupby(n.stores.loc[buses+ " Store", "bus"],axis=1).sum(axis=1)
-        #    stores[stores > 0.] = 0.
-        #    load += -stores
+    for carrier in carriers:
+        grouper = n.statistics.groupers.get_bus_and_carrier
+        balance = n.statistics.energy_balance(
```
Comment (p-glaum) [inline scripts/make_summary.py:548]: you can directely use withdrawal, then you do not need to wilter for values <0

File: scripts/make_summary.py [hunk 2]
```diff
@@ -540,77 +540,21 @@ def calculate_prices(n, label, prices):
...
+            groupby=grouper, aggregate_time=False, nice_names=False, bus_carrier=carrier
+        )
+        load = (balance[balance < 0].groupby(level=1).sum()).T
```
Comment (p-glaum) [inline scripts/make_summary.py:551]: better to do level="bus"

File: scripts/make_summary.py (hunk 1 above)
Comment (lisazeyen) [reply scripts/make_summary.py]: done!

File: scripts/make_summary.py (hunk 2 above)
Comment (lisazeyen) [reply scripts/make_summary.py]: Done!

**Comment by p-glaum** [comment]
> @p-glaum I am not dropping the prices if the load is zero at a bus. `price = price.reindex(columns=load.columns)` One could also reindex the other way around, but then the mariginal price would not be really load weighted, e.g. with `load.reindex(columns=price.columns, fill_value=1)`

okay, if the prices have a meaning if there is no load at the bus, then we should definitely keep them and weight every timestep equally, i.e. wit `load.reindex(columns=price.columns, fill_value=1)`
However, if they do not have a realistic interpretation, should we drop them?

**Review by p-glaum — APPROVED** [review]

**Comment by lisazeyen** [comment]
> > @p-glaum I am not dropping the prices if the load is zero at a bus. `price = price.reindex(columns=load.columns)` One could also reindex the other way around, but then the mariginal price would not be really load weighted, e.g. with `load.reindex(columns=price.columns, fill_value=1)`
> 
> okay, if the prices have a meaning for buses flows going in or out, then we should definitely keep them and weight every timestep equally, i.e. wit `load.reindex(columns=price.columns, fill_value=1)` However, if they do not have a realistic interpretation, should we drop them?

I think if it is load weighted and the load is zero we should drop them.

**Comment by fneum** [comment]
> I think if it is load weighted and the load is zero we should drop them.

I think equal weight would be slightly better, but dropping also ok.

**Review by fneum — APPROVED** [review]

**Pull request merged**